jedi/api/project.py

```diff
@@ -110,7 +110,17 @@ def _get_sys_path(self, inference_state, environment=None, add_parent_paths=True
                 suffixed += discover_buildout_paths(inference_state, inference_state.script_path)
 
                 if add_parent_paths:
-                    traversed = list(traverse_parents(inference_state.script_path))
+                    # Collect directories in upward search until we:
+                    #   1. Hit any directory without an __init__.py, AND
+                    #   2. Are above self._path.
+                    traversed = []
+                    no_init = False
+                    for parent_path in traverse_parents(inference_state.script_path):
+                        if not os.path.isfile(os.path.join(parent_path, "__init__.py")):
+                            no_init = True
```

Comment:
IMO it probably makes more sense to just:

```
if os.path.isfile(os.path.join(parent_path, "__init__.py")):
    continue
```

However that should be after the break (to avoid not hitting that). The `no_init` will not be needed anymore.

This is IMO the better solution, because sometimes people have python packages nested in weird ways (they compensate with path manipulation magic to get it working in the end).

What do you think?

Reply:
I see. So we avoid adding any directory to the path that does have an `__init__.py`.

I'll push an update and you can see what you think.

Reply:
The suggested implementation breaks a ton of integration tests. Maybe that's desired and the tests are wrong, but these tests are a bit beyond my understanding of the library at the moment. I'll push the implementation, but @davidhalter please check it. We can easily revert if you change your mind.